### PR TITLE
build: cover class methods and properties in api check

### PR DESCRIPTION
**Pull request by sobolk**
*Issue #, if available:*

**Description of changes:**

This PR adds coverage for class methods and properties in API check.
It also refactors how type check handles generic parameters to match what class and function generators do.

**Out of scope**
Other aspects of classes like inheritance and constructor. These will come separately.

**Example**

For API:
```typescript
export abstract class SomeAbstractClass {
    constructor(param1: string, param2?: string);
    someMethod: (param1: string, param2?: string) => string;
    someProperty: string;
    static someStaticMethod: (param1: string, param2?: string) => string;
    static someStaticProperty: string;
}
```

it generates:
```typescript
import { SomeAbstractClass } from 'samplePackageName';

const someAbstractClassSomeMethodUsageOuterFunction = (someAbstractClassSomeMethodUsageOuterFunctionParameter: SomeAbstractClass) => {
    const SomeAbstractClassSomeMethodUsageInnerFunction = (param1: string, param2?: string) => {
        const returnValue: string = someAbstractClassSomeMethodUsageOuterFunctionParameter.someMethod(param1);
        someAbstractClassSomeMethodUsageOuterFunctionParameter.someMethod(param1, param2);
    }
}
const someAbstractClassSomePropertyUsageOuterFunction = (someAbstractClassSomePropertyUsageOuterFunctionParameter: SomeAbstractClass) => {
    const propertyValue: string = someAbstractClassSomePropertyUsageOuterFunctionParameter.someProperty;
    
}
const someAbstractClassSomeStaticMethodUsageOuterFunction = (someAbstractClassSomeStaticMethodUsageOuterFunctionParameter: SomeAbstractClass) => {
    const SomeAbstractClassSomeStaticMethodUsageInnerFunction = (param1: string, param2?: string) => {
        const returnValue: string = SomeAbstractClass.someStaticMethod(param1);
        SomeAbstractClass.someStaticMethod(param1, param2);
    }
}
const someAbstractClassSomeStaticPropertyUsageOuterFunction = (someAbstractClassSomeStaticPropertyUsageOuterFunctionParameter: SomeAbstractClass) => {
    const propertyValue: string = SomeAbstractClass.someStaticProperty;
    
}
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by 0618 — APPROVED** [review]
LGTM

File: scripts/components/api-changes-validator/api_usage_statements_generators.ts [hunk 1]
```diff
@@ -209,12 +205,101 @@ export class ClassUsageStatementsGenerator implements UsageStatementsGenerator {
...
+        // it's fine to append duplicate ' | undefined' if type already has it.
+        propertyType += ' | undefined';
+      }
+      innerUsageStatement = `const propertyValue: ${propertyType} = ${callableSymbol};`;
+    }
+
+    let usageStatement = '';
+    usageStatement += `const ${outerUsageFunctionName} = ${genericTypeParametersDeclaration}(${outerUsageFunctionParameterName}: ${className}${genericTypeParameters}) => {${EOL}`;
+    usageStatement += `${addIndentation(innerUsageStatement)}${EOL}`;
+    usageStatement += `}${EOL}`;
```
[the remark this answers is not in the document]
Comment (sobolk) [reply scripts/components/api-changes-validator/api_usage_statements_generators.ts:298]: readability.
The way I have it in PR resembles how I would structure this code if I were writing it by hand.
I.e. enclosing function declaration and inner block are clearly visible.

File: scripts/components/api-changes-validator/api_usage_statements_generators.ts [hunk 2]
```diff
@@ -380,3 +469,14 @@ export class CallableParameterUsageStatementsGenerator
...
 };
+
+const toPascalCase = (name: string): string => {
+  return `${name?.substring(0, 1).toUpperCase()}${name?.substring(1)}`;
+};
+
+const addIndentation = (snippet: string): string => {
+  return snippet
+    .split(EOL)
+    .map((line) => `    ${line}`)
```
Comment (edwardfoyle) [inline scripts/components/api-changes-validator/api_usage_statements_generators.ts:480]: nit:
```suggestion
    .map((line) => `  ${line}`)
```

File: scripts/components/api-changes-validator/api_usage_statements_generators.ts [hunk 3]
```diff
@@ -380,3 +469,14 @@ export class CallableParameterUsageStatementsGenerator
 const toLowerCamelCase = (name: string): string => {
   return `${name?.substring(0, 1).toLowerCase()}${name?.substring(1)}`;
 };
+
+const toPascalCase = (name: string): string => {
+  return `${name?.substring(0, 1).toUpperCase()}${name?.substring(1)}`;
+};
+
+const addIndentation = (snippet: string): string => {
```
Comment (edwardfoyle) [inline scripts/components/api-changes-validator/api_usage_statements_generators.ts:477]: nit:
```suggestion
const indent = (snippet: string): string => {
```

**Review by edwardfoyle — APPROVED** [review]

**Review by edwardfoyle — APPROVED** [review]

**Review by 0618 — APPROVED** [review]

**Pull request merged**